custom_components/ttlock/api.py

```diff
@@ -203,3 +203,53 @@ async def set_passage_mode(self, lock_id: int, config: PassageModeConfig) -> boo
             return False
 
         return True
+
+    async def add_passcode(self, lock_id: int, config: AddPasscodeConfig) -> bool:
+        """Add new passcode."""
+
+        async with GW_LOCK:
+            res = await self.post(
+                "keyboardPwd/add",
+                lockId=lock_id,
+                addType=2,  # via gateway
+                keyboardPwd=config.passcode,
+                keyboardPwdName=config.passcode_name,
+                keyboardPwdType=3,  # Only temporary passcode supported
+                startDate=config.start_minute,
+                endDate=config.end_minute,
+            )
+
+        if "errcode" in res and res["errcode"] != 0:
+            _LOGGER.error("Failed to create passcode for %s: %s", lock_id, res["errmsg"])
+            return False
+
+        return True
+
+    async def delete_outdated_pass_codes(self, lock_id: int) -> bool:
+        """Get the list of pass codes of a lock."""
+        res = await self.get("lock/listKeyboardPwd", lockId=lock_id, pageNo=1, pageSize=100)
```

Comment:
It would be great if you could move this into it's own method (maybe `list_passcodes`) and add a model to wrap the response type, eg
```
async def list_passcodes(self, lock_id: int) -> list[Passcode]:
  ...
```